string[] sources
group_float poses
